test/e2e/autoscaler/autoscaler_test.go

```diff
@@ -0,0 +1,990 @@
+					queueCounts[queueAnnotation]++
+				}
+			}
+		}
+
+		// Verify pods are distributed across queues
+		Ω(len(queueCounts)).To(gomega.BeNumerically(">=", 2))
+	})
+
+	// Test autoscaling with preemption policies
+	It("Verify_Autoscaling_With_Preemption_Policies", func() {
+		By("Testing autoscaling behavior with different preemption policies (fence vs allow)")
+
+		fenceApp := "fence-app-" + common.RandSeq(5)
+		allowApp := "allow-app-" + common.RandSeq(5)
+		aggressorApp := "aggressor-app-" + common.RandSeq(5)
+
+		// Create fence policy deployment (cannot be preempted)
+		fenceDeployment := createTestDeployment(fenceApp, ns, 2, map[string]string{
+			"cpu":    "150m",
+			"memory": "192Mi",
+		})
+		fenceDeployment.Spec.Template.Annotations = map[string]string{
+			"yunikorn.apache.org/queue":             "root.fence-queue",
+			"yunikorn.apache.org/preemption-policy": "fence",
+		}
+
+		// Create allow policy deployment (can be preempted)
+		allowDeployment := createTestDeployment(allowApp, ns, 2, map[string]string{
+			"cpu":    "150m",
+			"memory": "192Mi",
+		})
+		allowDeployment.Spec.Template.Annotations = map[string]string{
+			"yunikorn.apache.org/queue":             "root.allow-queue",
+			"yunikorn.apache.org/preemption-policy": "allow",
+		}
+
+		// Deploy fence and allow policy apps
+		fenceDeploymentObj, err := kClient.CreateDeployment(fenceDeployment, ns)
+		Ω(err).NotTo(HaveOccurred())
+		defer func() { _ = kClient.DeleteDeployment(fenceDeploymentObj.Name, ns) }() //nolint:errcheck // Cleanup operation
+
+		allowDeploymentObj, err := kClient.CreateDeployment(allowDeployment, ns)
+		Ω(err).NotTo(HaveOccurred())
+		defer func() { _ = kClient.DeleteDeployment(allowDeploymentObj.Name, ns) }() //nolint:errcheck // Cleanup operation
+
+		By("Verifying initial deployment state")
+		gomega.Eventually(func() bool {
+			podsInNs, getErr := kClient.GetPods(ns)
+			if getErr != nil {
+				return false
+			}
+			fenceRunning := 0
+			allowRunning := 0
+
+			for _, pod := range podsInNs.Items {
+				if pod.Status.Phase == v1.PodRunning {
+					if strings.Contains(pod.Name, fenceApp) {
+						fenceRunning++
+					} else if strings.Contains(pod.Name, allowApp) {
+						allowRunning++
+					}
+				}
+			}
+
+			return fenceRunning >= 2 && allowRunning >= 2
+		}, 120*time.Second, 5*time.Second).Should(gomega.BeTrue())
+
+		By("Scaling up both deployments to consume more resources")
+		fenceDeployment.Spec.Replicas = &[]int32{4}[0]
+		allowDeployment.Spec.Replicas = &[]int32{4}[0]
+
+		_, err = kClient.UpdateDeployment(fenceDeployment, ns)
+		Ω(err).NotTo(HaveOccurred())
+		_, err = kClient.UpdateDeployment(allowDeployment, ns)
+		Ω(err).NotTo(HaveOccurred())
+
+		By("Waiting for scaled deployments")
+		time.Sleep(30 * time.Second) // Allow scaling to start
+
+		// Create high-priority aggressor deployment
+		aggressorDeployment := createTestDeployment(aggressorApp, ns, 3, map[string]string{
```

Comment:
Can't we call this `highPrioDeployment`? And in a similar vein, `highPrioApp`. The term "aggressor" sounds weird to me.